modules/oe_theme_content_news/oe_theme_content_news.module

```diff
@@ -33,6 +34,33 @@ function oe_theme_content_news_preprocess_node__oe_news(&$variables) {
   }
 }
 
+/**
+ * Implements template_preprocess_node() for the News node type.
+ */
+function oe_theme_content_news_preprocess_node__oe_news__teaser(&$variables) {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $variables['node'];
+
+  // Add news types to page metadata.
+  if (!$node->get('oe_news_types')->isEmpty()) {
+    $variables['metas']['types'] = _get_comma_separated_skos_meta($node->get('oe_news_types'));
+  }
+
+  // Add publication date to page metadata.
+  $timestamp = $node->get('oe_publication_date')->date->getTimestamp();
+  $variables['metas']['date'] = \Drupal::service('date.formatter')->format($timestamp, 'oe_theme_news_date');
+
+  // Add news locations to page metadata.
+  if (!$node->get('oe_news_location')->isEmpty()) {
+    $variables['metas']['location'] = _get_comma_separated_skos_meta($node->get('oe_news_location'));
+  }
+
+  // Add news authors to page metadata.
+  if (!$node->get('oe_author')->isEmpty()) {
+    $variables['metas']['authors'] = _get_comma_separated_skos_meta($node->get('oe_author'));
+  }
```

Comment:
As I understand it would be good to reorganize code for reducing code duplications everywhere https://github.com/openeuropa/oe_theme/blob/EWPP-614/modules/oe_theme_content_news/src/Plugin/PageHeaderMetadata/NewsContentType.php#L108.
Probably static method for building meta for new content type can be a good solution but it's only one of the solutions.
Anyway, it would be nice to have a single method/function for the formation of metas elements for rendering.